lhotse/cut.py

```diff
@@ -422,34 +423,38 @@ def truncate(
             recording=self.recording
         )
 
-    def pad(self, duration: Seconds, pad_feat_value: float = LOG_EPSILON) -> AnyCut:
+    def pad(
+            self,
+            duration: Seconds = None,
+            num_frames: int = None,
+            num_samples: int = None,
+            pad_feat_value: float = LOG_EPSILON,
+            direction: str = 'right'
+    ) -> AnyCut:
         """
-        Return a new MixedCut, padded to ``duration`` seconds with zeros in the recording,
-        and low-energy values in each feature bin.
+        Return a new MixedCut, padded with zeros in the recording, and ``pad_feat_value`` in each feature bin.
+
+        The user can choose to pad either to a specific `duration`; a specific number of frames `max_frames`;
+        or a specific number of samples `num_samples`. The three arguments are mutually exclusive.
 
+        :param cut: Cut to be padded.
         :param duration: The cut's minimal duration after padding.
+        :param num_frames: The cut's total number of frames after padding.
+        :param num_samples: The cut's total number of samples after padding.
         :param pad_feat_value: A float value that's used for padding the features.
             By default we assume a log-energy floor of approx. -23 (1e-10 after exp).
-        :return: a padded MixedCut if ``duration`` is greater than this cut's duration, otherwise ``self``.
+        :param direction: string, 'left' or 'right'. Determines whether the padding is added before or after
```

Comment:
Perhaps it would be useful to support 'both', or is there a reason why this isn't a good idea?

Reply:
.. 'both' could just recurse to the 'left' and 'right' forms if convenient.

Reply:
makes sense, will add